### PR TITLE
Separate Out CTC and SCC Proposer Roles

### PR DESCRIPTION
**Pull request by ben-chain**
## Description
Small PR to have separate addresses running the CTC and SCC.
## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: contracts/optimistic-ethereum/mockOVM/verification/mockOVM_BondManager.sol [hunk 1]
```diff
@@ -70,7 +70,7 @@ contract mockOVM_BondManager is iOVM_BondManager, Lib_AddressResolver {
         )
     {
         // Only authenticate sequencer to submit state root batches.
-        return _who == resolve("OVM_Sequencer");
+        return _who == resolve("OVM_Proposer");
```
Comment (ben-chain) [inline contracts/optimistic-ethereum/mockOVM/verification/mockOVM_BondManager.sol:73]: This is a notable redeploy requirement; could throw us for a loop if not careful @tynes

**Comment by maurelian** [comment]
Should we address #250 in this PR as well, or split them up?

**Review by tynes — APPROVED** [review]
I'm more in favor of this PR since its backward compatible

**Review by maurelian — APPROVED** [review]
LGTM, as long as @tynes and @ben-chain have communicated about this point: https://github.com/ethereum-optimism/contracts/pull/252#discussion_r583305465

File: contracts/optimistic-ethereum/mockOVM/verification/mockOVM_BondManager.sol (hunk 1 above)
Comment (smartcontracts) [reply contracts/optimistic-ethereum/mockOVM/verification/mockOVM_BondManager.sol:72]: Not a full redeploy though. Just need to change the address in the manager.

**Review by smartcontracts — APPROVED** [review]
LGTM. May have some downstream effects though. Let me take a look.

**Comment by smartcontracts** [comment]
Ok I can't really find any references to the sequencer address anywhere else: https://github.com/search?q=org%3Aethereum-optimism+OVM_Sequencer&type=code

**Pull request merged**